### PR TITLE
CON-3221 add api-acceptance cukes for trusted proxy support

### PR DESCRIPTION
**Pull request by apotterri**

### CODE REVIEW

File: acceptance-features/step_definitions/cli_steps.rb [hunk 1]
```diff
@@ -1,3 +1,23 @@
+Transform /\$ns/ do |s|
+  s.gsub('$ns', namespace)
```
Comment (dividedmind) [inline acceptance-features/step_definitions/cli_steps.rb]: Neat!

File: acceptance-features/step_definitions/http_steps.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+When /^I send a (GET|POST|PATCH|PUT|DELETE) request from "(.*?)" to "(.*?)"$/ do |method, ip, url|
+  headers['X-Forwarded-For'] = ip
+  step %Q{I send a #{method} request to "#{url}"}
+end
+
+When /^I send a (GET|POST|PATCH|PUT|DELETE) request from "(.*?)" to "(.*?)" with:$/ do |method, ip, url, params_table|
```
Comment (dividedmind) [inline acceptance-features/step_definitions/http_steps.rb]: I feel these steps should explicitly say `forwarded from`.

**Comment by dividedmind** [comment]
If I understand correctly, this relies on requests coming from localhost; which is to say, the tests would pass even without the trusted proxies patch. Am I missing something?

File: acceptance-features/step_definitions/http_steps.rb (hunk 2 above)
Comment (apotterri) [inline acceptance-features/step_definitions/http_steps.rb]: ok

**Comment by apotterri** [comment]
The tests rely on my trusted proxy changes being turned on, i.e. only localhost is regarded as a trusted proxy. We decided that this should be the default behavior. 

Without these changes, the tests all fail, complaining that the request was forwarded for `127.0.0.1`, instead of the expected address.

**Comment by dividedmind** [comment]
I was under the impression that 127.0.0.1 is a trusted proxy by default, cf. https://github.com/rack/rack/blob/master/lib/rack/request.rb#L419 -- is there some more magic to activating it?

**Comment by apotterri** [comment]
That's right, it is. But, if `TRUSTED_PROXIES` isn't set, you get the default rack behavior. Setting it to 127.0.0.1 turns on my changes, and ensures that all non-localhost IP addresses get used as-is.

**Comment by dividedmind** [comment]
Isn't the default rack behavior to trust 127.0.0.1?
22 sie 2016 16:40 "apotterri" <notifications@github.com> napisał(a):

**Comment by apotterri** [comment]
No, the default rack behavior maps all non-routable IP addresses to localhost.

**Pull request merged**